third-party/gtest/CMakeLists.txt

```diff
@@ -1,4 +1,3 @@
-find_package(GTest)
 if(NOT GTest_FOUND)
```

Comment:
Why keep the if()? Seems like it should be removed if we are not checking for the GTest package anymore.

Reply:
Addressed on [9324a29](https://github.com/The-OpenROAD-Project/OpenROAD/pull/3514/commits/9324a299b638ca02fa8ff443663732fe2fa1f219).